### PR TITLE
fixes #14855 - hide template type dropdown when snippet checked

### PR DESCRIPTION
**Pull request by domcleal**

### CODE REVIEW

File: app/assets/javascripts/editor.js [hunk 1]
```diff
@@ -103,7 +103,7 @@ function editor_file_source(evt){
 }
 
 function snippet_changed(item){
-  var checked = !!$(item).attr('checked');
+  var checked = $(item).is(':checked')
```
Comment (dLobatog) [inline app/assets/javascripts/editor.js]: Nitpick, missing semicolon

**Comment by dLobatog** [comment]
@domcleal :+1: thanks! I don't think the test failures are related - one of them is fixed on your other PR. Can you write an integration test to ensure we don't get the same regression again?

**Comment by domcleal** [comment]
Sure, I've added JS integration tests for snippets and templates to check the field hides/shows properly and that editing templates works. I've fixed the missing semicolon - thanks!

**Comment by dLobatog** [comment]
:+1: cool stuff, I'll merge if Mr Jenkins gives the OK

**Comment by dLobatog** [comment]
Some process got killed in jenkins, other than that is green. Merging !

http://ci.theforeman.org/job/test_develop_pr_core/9009/database=sqlite3,label=fast,ruby=2.0.0/console

**Comment by dLobatog** [comment]
Merged as deb71b8733b431d9637cbcdc889f994b930989bc, thanks @domcleal!

**Pull request closed without merging**